src/pages/Portfolio/LendingStats/BadgesList.vue

```diff
@@ -63,63 +67,30 @@ export default {
 			default: false
 		},
 	},
-	data() {
-		return {
-			challengeDataContentful: [],
-		};
-	},
 	computed: {
-		/* Challenge keys for completed challenges */
-		completedChallengeKeys() {
-			return this.completedAchievements.map(achievement => achievement.id).join(',');
-		},
 		inPortfolio() {
 			return this.$route?.name?.includes('portfolio');
 		},
 	},
-	watch: {
-		// When this prop has loaded we can load the badges content from contentful
-		totalPossibleBadges: {
-			handler(next) {
-				if (next !== 0) {
-					this.loadBadgesContent();
-				}
-			},
-			immediate: true
-		}
-	},
 	methods: {
-		loadBadgesContent() {
-			if (this.completedChallengeKeys !== '') {
-				const filterFields = `fields.key[in]=${this.completedChallengeKeys}`;
-				this.apollo.query({
-					query: gql`query contentfulDefinitions($customFields: String) {
-					contentful {
-						entries(contentType: "challenge", customFields: $customFields)
-					}
-				}`,
-					variables: {
-						customFields: filterFields
-					},
-				}).then(result => {
-					const contentfulData = result.data?.contentful?.entries?.items ?? null;
-					if (contentfulData) {
-						this.challengeDataContentful = contentfulData.map(item => {
-							return {
-								...item.fields,
-								badgeSvg: formatMediaAssetArray([item.fields.badgeImage])?.[0]?.file?.url,
-							};
-						});
-					}
-				}).catch(err => {
-					console.error(err);
-					this.$showTipMsg('There was a problem loading badges', 'error');
-				});
+		getBadgeTitle(badge) {
+			if (badge.level === 0) {
+				return badge?.contentfulData?.[0]?.challengeName ?? '';
 			}
+			const badgeData = badge?.contentfulData?.find(data => data.level === badge.level);
+			return `${badgeData?.challengeName} ${badgeData?.levelName}` ?? '';
 		},
-		shouldShowBadgeDate(challenge) {
-			// Lifetime badges currently have an "n/a" date tagline
-			return challenge?.dateTagline?.toLowerCase() !== 'n/a';
+		getBadgeImgUrl(badge) {
+			if (badge.level === 0) {
+				return badge?.contentfulData?.[0]?.imageUrl ?? '';
+			}
+			const badgeData = badge?.contentfulData?.find(data => data.level === badge.level);
+			return badgeData?.imageUrl ?? '';
+		},
+
+		getBadgeDate(badge) {
```

Comment:
It looks like this new method is missing some logic from `shouldShowBadgeDate`, do we need the `n/a` check?